_posts/2017-05-15-rust-at-two-years.md

```diff
@@ -0,0 +1,352 @@
+the second edition has boarded [the beta train] and will be an option in the
+documentation shipped with Rust 1.18.0. Steve and I have gotten feedback that
+[the ownership chapter] especially is much improved and has helped people
+understand ownership related concepts better!
+
+[The Rust Programming Language Book]: https://github.com/rust-lang/book
+[preorder from No Starch]: https://www.nostarch.com/rust
+[the beta train]: https://doc.rust-lang.org/beta/book/
+[the ownership chapter]: https://doc.rust-lang.org/nightly/book/second-edition/ch04-00-understanding-ownership.html
+
+The [Language Ergonomics Initiative] is another part of the lower learning
+curve goal that has [a number of improvements][lei-tracker] in its pipeline.
+The language team is eager to mentor people (another goal!) who are interested
+in getting involved with moving these ergonomic improvement ideas forward by
+writing RFCs and working with the community to flesh out the details of how
+these improvements would work. Comment on the [tracking issue][lei-tracker] if
+you'd like to jump in!
+
+[Language Ergonomics Initiative]: https://blog.rust-lang.org/2017/03/02/lang-ergonomics.html
+[lei-tracker]: https://github.com/rust-lang/rust-roadmap/issues/17
+
+Also check out:
+
+- [The Rust Cookbook](https://github.com/brson/rust-cookbook)
+- [The new error format rolled out in Rust 1.12.0](https://github.com/rust-lang/rust/issues/35233)
+- [The question mark operator stabilized in Rust 1.13.0](https://github.com/rust-lang/rust/pull/31954)
+
+### [Rust should have a pleasant edit-compile-debug cycle](https://github.com/rust-lang/rust-roadmap/issues/1)
+
+Compiler improvements are ongoing. This year, [MIR became a default part of the
+compilation process](https://github.com/rust-lang/rust/pull/34096), which was a
+prerequisite to unlocking further improvements.
+
+Below is a table of [benchmarks] comparing the time it takes to compile a few
+crates and tests with Rust 1.8.0 (the stable release a year ago) to Rust 1.17.0
+(the stable release today). On average, compile times have improved by 5-10%,
+but some worst-case behavior has been fixed that results in >95% improvements
+in certain programs. A couple of crates did show a slight regression of 1-15%
+in compile times, but in most cases it's a win. The helloworld crate saw a 110%
+increase in compile time, but it only equates to +0.1 second, which is probably
+a small fixed overhead that gets overwhelmed in any larger project.
+
+| Benchmark | 1.8.0 (time) | 1.8.0 ([rss]) | 1.17.0 (time) | 1.17.0 ([rss]) | % change (time) | % change ([rss]) |
+|-------|--------------|---------------|---------------|----------------|-----------------|------------------|
+| hyper 0.5.0 | 5.97s | 248MB | 5.44s | 234MB | -8.9% | -5.6 % |
+| html5ever 2016-08-25 | 5.16s | 234MB | 3.98s | 228MB | -22.9% | -2.6% |
+| issue-32062 | 10.40s | 127MB | 0.26s | 89MB | -97.5% | -29.9% |
```

Comment:
I think I would remove the `issue-XXX` rows. They don't look like "real code" (because they're not), and they don't show anything extraordinary (in particular, I think that the gains here are also reflected in jld-day15-parser, although that may be coincidence, I can't remember if those are due to a similar cause).